homeassistant/components/trend/binary_sensor.py

```diff
@@ -117,20 +120,22 @@ class SensorTrend(BinarySensorEntity):
     """Representation of a trend Sensor."""
 
     _attr_should_poll = False
+    _gradient = 0.0
```

Comment:
Why do we change this from `None`?

Reply:
I changed this from `None` because `mypy` complains about incompatible types for `abs` and `copysign`.

Reply:
I didn't scan the code but does it change any logic?

Reply:
Not from my point of view. `_gradient` is used only in this part of the code:

```python
self._state = (
    abs(self._gradient) > abs(self._min_gradient)
    and math.copysign(self._gradient, self._min_gradient) == self._gradient
)
```

Reply:
Right, it actually calculates it before that code you pasted so the starting value could be anything really.